src/scf/self_consistent_field.jl

```diff
@@ -137,37 +129,43 @@ Overview of parameters:
             energies = energy_hamiltonian(basis, ψ, occupation;
                                           ρ=ρout, eigenvalues, εF).energies
         end
-        info = merge(info, (; energies))
+        info_next = merge(info_next, (; energies))
 
         # Apply mixing and pass it the full info as kwargs
-        δρ = mix_density(mixing, basis, ρout - ρin; info...)
-        ρnext = ρin .+ T(damping) .* δρ
-        info = merge(info, (; ρnext))
+        Δρ = mix_density(mixing, basis, ρout - ρin; info_next...)
+        ρnext = ρin .+ T(damping) .* Δρ
+        info_next = merge(info_next, (; ρnext))
 
-        callback(info)
-        is_converged(info) && (converged = true)
+        callback(info_next)
+        if is_converged(info_next)
+          info_next = merge(info_next, (; converged=true))
+        end
 
-        ρnext
+        ρnext, info_next
     end
 
+    info_init = (; ρin=ρ, ψ=ψ, occupation=nothing, eigenvalues=nothing, εF=nothing, 
+                   n_iter=0, converged=false)
+
     # Tolerance and maxiter are only dummy here: Convergence is flagged by is_converged
     # inside the fixpoint_map.
-    solver(fixpoint_map, ρout, maxiter; tol=eps(T))
-
+    _, info = solver(fixpoint_map, ρ, info_init, maxiter; tol=eps(T)) # TODO ?? why dummy?
```

Comment:
On that TODO: I think I understand now the comment above for custom convergence flagging - I wonder whether we could move this to the fixed-point solvers API though? instead of specifying the tolerance, one could do something like `solver(fixpoint_map, ρ, info_init, maxiter; is_converged_fn=is_converged_fn)`?

Reply:
We can very much remove this dummy stuff now !

Reply:
Done in [caee55e](https://github.com/JuliaMolSim/DFTK.jl/pull/811/commits/caee55e87e9eb0a1c9b487d3c25daca60a312b10)